.pre-commit-config.yaml

```diff
@@ -138,3 +138,14 @@ repos:
       - id: renovate-config-validator
         args:
           - --strict
+
+  - repo: local
+    hooks:
+      - id: codeowners-validator
+        name: CODEOWNERS validator
+        language: docker_image
+        pass_filenames: false
+        entry: >
+          -e REPOSITORY_PATH="."
+          -e CHECKS="files,duppatterns,syntax"
+          ghcr.io/mszostok/codeowners-validator:v0.7.4
```

Comment:
It'd be nice to include the checks for `notowned` and `avoid-shadowing` once they pass the experimental phase. Also, the description of the latter suggest than the `*` pattern may work as we want if it's specified before the more specific line but I haven't confirmed it 🤔

Reply:
The `notowned` check requires the branch to not have any uncommited changes, and so cannot run in our normal `pre-commit` stage. It's flaky to run it in `pre-push` as well because it will still complain if there are uncommitted changes (like you have something you would have otherwise stashed).

It's a good check but not currently easily usable. We'd need to add a separate step for it altogether, probably just in CI. I couldn't find an easy way to do that in pre-commit.

Reply:
Thanks for enumerating the quirks of the `notowned` check. I assumed there would be some complications, but nothing so disrupting 🤯  Anyway, my comment here was just a desire (though at first, I thought it was the whole reason to add it to pre-commit); I wasn't requesting changes for this to be clear.